docs/DeveloperGuide.md

```diff
@@ -1,38 +1,584 @@
+> check the index provided using `Parser#checkValidCategory` before iterating `Family`. If the check
+> fails, an error message will be displayed to the user instead of continuing with the execution.
+
+The following sequence diagram shows how the `listcat` operation works after a `ListCatCommand`
+command object has been created by [`CommandFactory`](#Command-Execution):
+
+![ListCategorisedExpenseSequence](images/ListCategorisedExpenseSequence.png)
+
+#### Design considerations:
+
+**Aspect: How to categorise expenses to be listed:**
+
+* **Alternative 1 (current choice):** Adding a category attribute to expenses.
+    * Pros: Easy to implement and less memory usage.
+    * Cons: May have performance issues as it needs to iterate through every person's expenditure.
+
+* **Alternative 2:** Maintain an array list for each category and store a copy of expenses.
+    * Pros: Fast to print expenses in a category, no unnecessary look-ups.
+    * Cons: Poor memory management, needs to store twice as many expenditures.
+
+### Data Archiving
+
+#### Saving and loading feature
+
+##### Implementation
+
+The saving and loading mechanism is facilitated by `Family`. It loads records of `PersonList` for each logical
+grouping inside family from a local file, process the data and adds the record to `Family` for the current session.
+Saves the `PersonList` record of each `Family` grouping back into the local file upon exit of the program. It implements
+the following operations:
+
+* `Storage#checkFileExists()` -- Checks if `PlanITarium.txt` file exists in the local hard drive, creates one if it does
+  not.
+* `Storage#saveData()` -- Writes `PersonList` record of each `Family` grouping into `PlanITarium.txt` file upon exit of
+  the program.
+* `Storage#loadData()` -- Opens and reads `PlanITarium.txt` for any existing `Family` records and updates each
+  `PersonList` data in the current session for each logical group in `Family`.
+
+Given below is an example scenario and how the save and load mechanism behaves at each step.
+
+Step 1. The user launches the application. A `Storage` object will be initialised with an empty `PersonList` for each
+`Family` grouping. The following sequence diagram shows how the `Storage` object is initialised
+
+Step 2. The program will create a new `Family` object for the session which calls `Storage#loadData()`, causing a check 
+if the local file `PlanITarium.txt` exists by calling `Storage#checkFileExists()` and creates one if it does not. The 
+data in the local file will be read, parsed and added to the empty `Family` in `Storage` by calling `Family` adding
+operations. After all lines of the files has been read and all data added, the `Family` object is then returned to main
+and the `Family` object of the current session will consist of the data in `Family` of `Storage`. 
+The following sequence diagram shows how the loading operation works:
+
+![StorageLoadSequence](images/StorageLoadSequence.png)
```

Comment:
The diagram seems unnecessarily wide? Maybe could break the method calls into smaller chunks, and not write them into same line so that the diagram can be smaller?